### PR TITLE
#34 ZipSlice implemented

### PR DESCRIPTION
**Pull request by olenagerasimova**
For #34 I've implemented `ZipSlice` and reorganized / removed classes to correspond current structure.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#37](https://github.com/artipie/go-adapter/issues/37) is assigned to @hdouss[/z](https://www.0crat.com/u/hdouss), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/67eb5964-3b19-4a71-9b88-f2ca334e0b02); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by HDouss** [comment]
@olenagerasimova This does not seem related to #35 . Please correct the associated issue.

**Comment by olenagerasimova** [comment]
@HDouss yes, that's actually for #34, sorry

File: src/main/java/com/artipie/http/ZipSlice.java [hunk 1]
```diff
@@ -23,25 +23,45 @@
...
     public Response response(
         final String line, final Iterable<Map.Entry<String, String>> headers,
-        final Publisher<ByteBuffer> body) {
-        throw new NotImplementedException("Not yet implemented");
+        final Publisher<ByteBuffer> body
+    ) {
+        return new RsWithHeaders(
+            this.origin.response(line, headers, body),
+            new ListOf<Map.Entry<String, String>>(
+                new MapEntry<>("content-type", "application/zip")
```
Comment (HDouss) [inline src/main/java/com/artipie/http/ZipSlice.java:63]: @olenagerasimova may be that is unrelated to the issue, but I think the class name is confusing, the only "decorating effect" here is to add a header, nothing related to "zip". Either a further implementation required and a puzzle is required, or the class should be more generic: something like "SliceWithHeaders", decorating the response of an origin slice with a passed header.

**Comment by olenagerasimova** [comment]
@HDouss thanks, I've made `ZipSlice` to wrap `DownloadSlice` (in this repo we will need it often). Also noticed that in the test wrong `Slice` instance (I'm embraced) was created and `RsHasHeaders` works fine.

**Comment by olenagerasimova** [comment]
@HDouss I see your point and I agree, but this repository is not the right place for proper `SliceWithHeaders` decorator, it should be added in artipie-http. If we agree to do it, then I'll create issue here to get rid of `DownloadWithCntTypeSlice` and use "official" decorator.

**Comment by HDouss** [comment]
@g4s8 Please review or merge.

**Comment by g4s8** [comment]
@olenagerasimova I agree with @HDouss - we can move `SliceWithHeaders` to http module and use it here

**Comment by olenagerasimova** [comment]
@g4s8 created PR for `SliceWithHeaders`  https://github.com/artipie/http/pull/134

**Comment by HDouss** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/artipie/go-adapter/issues/37#issuecomment-618595175))

@HDouss This is what I know about this job in CT2E6TK9B, as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#37](https://github.com/artipie/go-adapter/issues/37) is [in scope](http://www.zerocracy.com/policy.html#14) for 2days
  * The role is `REV`
  * The job is assigned to @hdouss[/z](https://www.0crat.com/u/hdouss) for [2days](http://www.zerocracy.com/policy.html#8)
  * There is a monetary reward attached
  * @hdouss[/z](https://www.0crat.com/u/hdouss) will get 25% of their hourly rate for this job
  * The job doesn't have any [impediments](http://www.zerocracy.com/policy.html#9)
  * The [budget](http://www.zerocracy.com/policy.html#4) is 15 minutes/points
  * These users are banned and won't be assigned:
    * @olenagerasimova[/z](https://www.0crat.com/u/olenagerasimova): This user reported the ticket
  * Job [footprint](https://www.0crat.com/footprint/CT2E6TK9B?q=%7Bjob%3A%27gh%3Aartipie%2Fgo-adapter%2337%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by HDouss** [comment]
@g4s8 how to trigger rewards here?

**Comment by g4s8** [comment]
@0crat out

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @hdouss[/z](https://www.0crat.com/u/hdouss), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@0crat quality good

**Pull request merged**